src/basic/fileio.c

```diff
@@ -300,6 +301,9 @@ int read_full_stream_full(
                          * already makes us notice the EOF. */
                         if (st.st_size > 0)
                                 n_next = st.st_size + 1;
+
+                        if (secure)
+                                (void) warn_file_is_world_accessible(fn, &st, NULL, 0);
```

Comment:
if we had a flags param, then this could also be protected by a separate flag maybe?